interface/patient_file/summary/insurance_edit.php

```diff
@@ -0,0 +1,114 @@
+
+// make sure permissions are checked before we allow this page to be accessed.
+if (!AclMain::aclCheckCore('patients', 'demo', '', 'write')) {
+    die(xlt('Updating demographics is not authorized.'));
+}
+
+// Session pid must be right or bad things can happen when demographics are saved!
+//
+$set_pid = $_GET["set_pid"] ?? ($_GET["pid"] ?? null);
+if ($set_pid && $set_pid != $_SESSION["pid"]) {
+    setpid($set_pid);
+}
+
+$result = getPatientData($pid, "*, DATE_FORMAT(DOB,'%Y-%m-%d') as DOB_YMD");
+
+ // Check authorization.
+if ($pid) {
+    // Create and fire the patient demographics update event
+    $updateEvent = new UpdateEvent($pid);
+    $updateEvent = $GLOBALS["kernel"]->getEventDispatcher()->dispatch($updateEvent, UpdateEvent::EVENT_HANDLE, 10);
+
+    if (
+        !$updateEvent->authorized() ||
+        !AclMain::aclCheckCore('patients', 'demo', '', 'write')
+    ) {
+        die(xlt('Updating insurance is not authorized.'));
+    }
+
+    if ($result['squad'] && ! AclMain::aclCheckCore('squads', $result['squad'])) {
+        die(xlt('You are not authorized to access this squad.'));
+    }
+} else {
+    if (!AclMain::aclCheckCore('patients', 'demo', '', array('write','addonly'))) {
+        die(xlt('Adding insurance is not authorized.'));
+    }
+}
+// $statii = array('married','single','divorced','widowed','separated','domestic partner');
+// $provideri = getProviderInfo();
+if ($GLOBALS['insurance_information'] != '0') {
+    $insurancei = getInsuranceProvidersExtra();
+} else {
+    $insurancei = getInsuranceProviders();
+}
+//Check to see if only one insurance is allowed
+if ($GLOBALS['insurance_only_one']) {
+    $insurance_array = array('primary');
+} else {
+    $insurance_array = array('primary', 'secondary', 'tertiary');
+}
+
+//Check to see if only one insurance is allowed
+if ($GLOBALS['insurance_only_one']) {
+    $insurance_headings = array(xl("Primary Insurance Provider"));
+} else {
+    $insurance_headings = array(xl("Primary Insurance Provider"), xl("Secondary Insurance Provider"), xl("Tertiary Insurance provider"));
+}
+
+$twig = (new TwigContainer(null, $GLOBALS['kernel']))->getTwig();
+//$insurance_info[0]['active'] = true;
+//$insuranceTypes = array_map(function($item) { return $item['type'];}, $insurance_info);
+//$insrender(uranceTypes = array_unique($insuranceTypes);
+// we DO NOT want to allow users to add states/localities in this form per Business Rules
+// so if we have a state add button we are removing it here.
+$state_data_type = $GLOBALS['state_data_type'] === '26' ? '1' : $GLOBALS['state_data_type'];
+$country_data_type = $GLOBALS['country_data_type'] === '26' ? '1' : $GLOBALS['country_data_type'];
+echo $twig->render(
+    "patient/insurance/insurance_edit.html.twig",
+    [
+        'insuranceTypes' => $insurance_array,
+        'activeType' => $insurance_array[0],
+        'patient' => $result,
+        'puuid' => \OpenEMR\Common\Uuid\UuidRegistry::uuidToString($result['uuid'])
```

Comment:
abbreviate class path and put in above `use block`

Reply:
Fixed